modules/EvseManager/tests/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+set(TEST_TARGET_NAME ${PROJECT_NAME}_EvseManager_tests)
+add_executable(${TEST_TARGET_NAME})
+
+target_include_directories(${TEST_TARGET_NAME} PUBLIC
+    ${GTEST_INCLUDE_DIRS} . ..
+    ${GENERATED_INCLUDE_DIR}
+    ${CMAKE_BINARY_DIR}/generated/modules/${MODULE_NAME}
+    ${CMAKE_BINARY_DIR}/generated/include
+)
+
+target_sources(${TEST_TARGET_NAME} PRIVATE
+    EnumFlagsTest.cpp
+    ErrorHandlingTest.cpp
+    EventQueueTest.cpp
+    ../ErrorHandling.cpp
+)
+
+target_compile_definitions(${TEST_TARGET_NAME} PRIVATE
+    BUILD_TESTING_MODULE_EVSE_MANAGER
+)
+
+
+find_package(GTest REQUIRED)
```

Comment:
This find_package is not needed anymore

Reply:
it fails to build for me without it. Many link errors.

Reply:
If you replace the GTEST_LIBRARIES and GTEST_MAIN_LIBRARIES in the target_link_libraries with
 GTest::gtest_main
It should work, [this is done like this in other unit tests as well](https://github.com/EVerest/everest-core/blob/aed024ba594d9842dacbc25355407c03ba7bffa7/modules/Auth/tests/CMakeLists.txt#L17C4-L17C22)